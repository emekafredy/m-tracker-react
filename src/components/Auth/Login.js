import React, { Component } from 'react';
import classnames from 'classnames';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

// action
import { loginUser } from '../../actions/auth';

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      errors: {}
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    if (this.props.auth.isAuthenticated) {
      this.props.history.push('/requests');
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.auth.isAuthenticated) {
      this.props.history.push('/requests');
    }
    if (nextProps.errors.errors) {
      this.setState({ errors: nextProps.errors.errors })
    }
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value })
  }

  handleSubmit(event) {
    event.preventDefault();
    
    const { loginUser } = this.props;

    const registeredUser = {
      email: this.state.email,
      password: this.state.password
    }

    loginUser(registeredUser);
  }

  render() {
    const errors = this.state;
    return (
      <div className="signUp">
        <div className="card">
          <form className="auth-form" onSubmit={ this.handleSubmit }> 
            <div className="centre-div">
              <h1> Login </h1>
            </div>
            <input
              className={classnames('input-class', {
                'error-outline': errors.email
              })}
              id="email"
              type="text" placeholder="Email" name="email"
              value={ this.state.email }
              onChange={ this.handleChange }
            />
            { errors.errors.email ? (<div className="error-message">{ errors.errors.email }</div>) : '' }

            <input
              className={classnames('input-class', {
                'error-outline': errors.password
              })}
              id="password"
              type="password" placeholder="Password" name="password"
              value={ this.state.password }
              onChange={ this.handleChange }
            />
            { errors.errors.password ? (<div className="error-message">{ errors.errors.password }</div>) : '' }

            <input type="submit" value="Login" id="submit"/> 

          </form>
          <div className="centre-div">
            <p>Don't have an account ?</p>
            <Link className="links" to="/register"> Sign up </Link>
          </div>
        </div>
      </div>
    )
  }
}

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
}

const mapStateToProps = (state) => ({
  auth: state.auth,
  errors: state.errors,
});

export default connect(mapStateToProps, { loginUser })(withRouter(Login));