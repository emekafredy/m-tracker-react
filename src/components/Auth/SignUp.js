import React, { Component } from 'react';
import classnames from 'classnames';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

// action
import { registerUser } from '../../actions/auth';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      errors: {}
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    if (this.props.auth.isAuthenticated) {
      this.props.history.push('/requests');
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.auth.isAuthenticated) {
      this.props.history.push('/requests');
    }
    if (nextProps.errors.errors) {
      this.setState({ errors: nextProps.errors.errors })
    }
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value })
  }

  handleSubmit(event) {
    event.preventDefault();

    const { registerUser } = this.props;
    
    const newUser = {
      firstName: this.state.firstName,
      lastName: this.state.lastName,
      email: this.state.email,
      password: this.state.password
    }

    registerUser(newUser);
  }

  render() {
    const { errors } = this.state;
    return (
      <div className="signUp">
        <div className="card">
          <form className="auth-form" onSubmit={ this.handleSubmit }> 
            <div className="centre-div">
              <h2> Sign up </h2>
              <p>Please fill out every field</p>
            </div>
            <input 
              className={classnames('input-class', {
                'error-outline': errors.firstName
              })}
              id="firstName"
              type="text" placeholder="First name" name="firstName"
              value={ this.state.firstName }
              onChange={ this.handleChange }
            />
            { errors.firstName ? (<div className="error-message">{ errors.firstName }</div>) : '' }

            <input
            className={classnames('input-class', {
              'error-outline': errors.lastName
            })}
            id="lastName"
            type="text" placeholder="Last name" name="lastName"
            value={ this.state.lastName }
            onChange={ this.handleChange }
            />
            { errors.lastName ? (<div className="error-message">{ errors.lastName }</div>) : '' }
      
            <input
            className={classnames('input-class', {
              'error-outline': errors.email
            })}
            id="email"
            type="text" placeholder="Email" name="email"
            value={ this.state.email }
            onChange={ this.handleChange }
            />
            { errors.email ? (<div className="error-message">{ errors.email }</div>) : '' }

            <input
            className={classnames('input-class', {
              'error-outline': errors.password
            })}
            id="password"
            type="password" placeholder="Password" name="password"
            value={ this.state.password }
            onChange={ this.handleChange }
            />
            { errors.password ? (<div className="error-message">{ errors.password }</div>) : '' }

            <input type="submit" value="Register" id="register"/>
          </form>
          <div className="centre-div">
            <p>Already Registered ?</p>
            <Link className="links" to="/login"> Login </Link>
          </div>
        </div>
      </div>
    )
  }
}

SignUp.propTypes = {
  registerUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
}

const mapStateToProps = (state) => ({
  auth: state.auth,
  errors: state.errors,
});

export default connect(mapStateToProps, { registerUser })(withRouter(SignUp));
