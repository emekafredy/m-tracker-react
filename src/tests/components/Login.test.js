import React from 'react';
import { shallow } from 'enzyme';
import { loginDetails } from '../__mocks__/mockData';

import { Login } from '../../components/Auth/Login';

let wrapper;

describe('Login component', () => {

  beforeEach(() => {
    const props = {
      action: {
        loginUser: jest.fn()
      },
      errors: {},
      auth: {
        isAuthenticated: true
      },
      history: {
        push: jest.fn()
      }
    };
    wrapper = shallow(
        <Login {...props} />
    );
  });

  test("renders the Login component", () => { 
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  test('it sets the email when its state changes', () => {
    const event = {
      target: {
        name: 'email',
        value: 'emeka@mail.com'
      }
    };

    wrapper.find('#email').simulate('change', event);;
    const expectedResult = 'emeka@mail.com';

    expect(wrapper.instance().state.email).toBe(expectedResult);
  });

  test('it sets the password when its state changes', () => {
    const event = {
      target: {
        name: 'password',
        value: 'mypass'
      }
    };

    wrapper.find('#password').simulate('change', event);
    const expectedResult = 'mypass';

    expect(wrapper.instance().state.password).toBe(expectedResult);
  });

  test('it successfully logs in a registered user', () => {
    const event = {
      preventDefault: jest.fn(),
    };

    const login = wrapper.find('#submit');
    wrapper.setState(loginDetails);
    login.simulate('click', event);
  
    expect(wrapper.instance().state.errors).toEqual({});
  });

  test('it fails when user makes a bad request', () => {
    const event = {
      preventDefault: jest.fn(),
    };

    const login = wrapper.find('#submit');
    login.simulate('click', event);
    expect(wrapper.instance().state.errors).not.toBe({});
  });

  test('it fails when user makes a bad request', () => {
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'email',
        value: ''
      }
    };

    wrapper.find('#submit').simulate('click', event);

    expect(wrapper).toMatchSnapshot();
  });
  
})
