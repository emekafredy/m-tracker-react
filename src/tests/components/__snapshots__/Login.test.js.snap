// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component it fails when user makes a bad request 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    action={
      Object {
        "loginUser": [MockFunction],
      }
    }
    auth={
      Object {
        "isAuthenticated": true,
      }
    }
    errors={Object {}}
    history={
      Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/requests",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card"
      >
        <form
          className="auth-form"
          onSubmit={[Function]}
        >
          <div
            className="centre-div"
          >
            <h1>
               Login 
            </h1>
          </div>
          <input
            className="input-class"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
          
          <input
            className="input-class"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          
          <input
            id="submit"
            type="submit"
            value="Login"
          />
        </form>
        <div
          className="centre-div"
        >
          <p>
            Don't have an account ?
          </p>
          <Link
            className="links"
            replace={false}
            to="/register"
          >
             Sign up 
          </Link>
        </div>
      </div>,
      "className": "signUp",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="auth-form"
            onSubmit={[Function]}
          >
            <div
              className="centre-div"
            >
              <h1>
                 Login 
              </h1>
            </div>
            <input
              className="input-class"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />
            
            <input
              className="input-class"
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            
            <input
              id="submit"
              type="submit"
              value="Login"
            />
          </form>,
          <div
            className="centre-div"
          >
            <p>
              Don't have an account ?
            </p>
            <Link
              className="links"
              replace={false}
              to="/register"
            >
               Sign up 
            </Link>
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="centre-div"
              >
                <h1>
                   Login 
                </h1>
              </div>,
              <input
                className="input-class"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
              />,
              "",
              <input
                className="input-class"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />,
              "",
              <input
                id="submit"
                type="submit"
                value="Login"
              />,
            ],
            "className": "auth-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                   Login 
                </h1>,
                "className": "centre-div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Login ",
                },
                "ref": null,
                "rendered": " Login ",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-class",
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-class",
                "id": "password",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "submit",
                "type": "submit",
                "value": "Login",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Don't have an account ?
              </p>,
              <Link
                className="links"
                replace={false}
                to="/register"
              >
                 Sign up 
              </Link>,
            ],
            "className": "centre-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Don't have an account ?",
              },
              "ref": null,
              "rendered": "Don't have an account ?",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " Sign up ",
                "className": "links",
                "replace": false,
                "to": "/register",
              },
              "ref": null,
              "rendered": " Sign up ",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card"
        >
          <form
            className="auth-form"
            onSubmit={[Function]}
          >
            <div
              className="centre-div"
            >
              <h1>
                 Login 
              </h1>
            </div>
            <input
              className="input-class"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />
            
            <input
              className="input-class"
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            
            <input
              id="submit"
              type="submit"
              value="Login"
            />
          </form>
          <div
            className="centre-div"
          >
            <p>
              Don't have an account ?
            </p>
            <Link
              className="links"
              replace={false}
              to="/register"
            >
               Sign up 
            </Link>
          </div>
        </div>,
        "className": "signUp",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <form
              className="auth-form"
              onSubmit={[Function]}
            >
              <div
                className="centre-div"
              >
                <h1>
                   Login 
                </h1>
              </div>
              <input
                className="input-class"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
              />
              
              <input
                className="input-class"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              
              <input
                id="submit"
                type="submit"
                value="Login"
              />
            </form>,
            <div
              className="centre-div"
            >
              <p>
                Don't have an account ?
              </p>
              <Link
                className="links"
                replace={false}
                to="/register"
              >
                 Sign up 
              </Link>
            </div>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="centre-div"
                >
                  <h1>
                     Login 
                  </h1>
                </div>,
                <input
                  className="input-class"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="text"
                  value=""
                />,
                "",
                <input
                  className="input-class"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />,
                "",
                <input
                  id="submit"
                  type="submit"
                  value="Login"
                />,
              ],
              "className": "auth-form",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1>
                     Login 
                  </h1>,
                  "className": "centre-div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Login ",
                  },
                  "ref": null,
                  "rendered": " Login ",
                  "type": "h1",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-class",
                  "id": "email",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Email",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-class",
                  "id": "password",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "submit",
                  "type": "submit",
                  "value": "Login",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Don't have an account ?
                </p>,
                <Link
                  className="links"
                  replace={false}
                  to="/register"
                >
                   Sign up 
                </Link>,
              ],
              "className": "centre-div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Don't have an account ?",
                },
                "ref": null,
                "rendered": "Don't have an account ?",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": " Sign up ",
                  "className": "links",
                  "replace": false,
                  "to": "/register",
                },
                "ref": null,
                "rendered": " Sign up ",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Login component renders the Login component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    action={
      Object {
        "loginUser": [MockFunction],
      }
    }
    auth={
      Object {
        "isAuthenticated": true,
      }
    }
    errors={Object {}}
    history={
      Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/requests",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card"
      >
        <form
          className="auth-form"
          onSubmit={[Function]}
        >
          <div
            className="centre-div"
          >
            <h1>
               Login 
            </h1>
          </div>
          <input
            className="input-class"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
          
          <input
            className="input-class"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          
          <input
            id="submit"
            type="submit"
            value="Login"
          />
        </form>
        <div
          className="centre-div"
        >
          <p>
            Don't have an account ?
          </p>
          <Link
            className="links"
            replace={false}
            to="/register"
          >
             Sign up 
          </Link>
        </div>
      </div>,
      "className": "signUp",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="auth-form"
            onSubmit={[Function]}
          >
            <div
              className="centre-div"
            >
              <h1>
                 Login 
              </h1>
            </div>
            <input
              className="input-class"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />
            
            <input
              className="input-class"
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            
            <input
              id="submit"
              type="submit"
              value="Login"
            />
          </form>,
          <div
            className="centre-div"
          >
            <p>
              Don't have an account ?
            </p>
            <Link
              className="links"
              replace={false}
              to="/register"
            >
               Sign up 
            </Link>
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="centre-div"
              >
                <h1>
                   Login 
                </h1>
              </div>,
              <input
                className="input-class"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
              />,
              "",
              <input
                className="input-class"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />,
              "",
              <input
                id="submit"
                type="submit"
                value="Login"
              />,
            ],
            "className": "auth-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                   Login 
                </h1>,
                "className": "centre-div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Login ",
                },
                "ref": null,
                "rendered": " Login ",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-class",
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-class",
                "id": "password",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "submit",
                "type": "submit",
                "value": "Login",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Don't have an account ?
              </p>,
              <Link
                className="links"
                replace={false}
                to="/register"
              >
                 Sign up 
              </Link>,
            ],
            "className": "centre-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Don't have an account ?",
              },
              "ref": null,
              "rendered": "Don't have an account ?",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " Sign up ",
                "className": "links",
                "replace": false,
                "to": "/register",
              },
              "ref": null,
              "rendered": " Sign up ",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card"
        >
          <form
            className="auth-form"
            onSubmit={[Function]}
          >
            <div
              className="centre-div"
            >
              <h1>
                 Login 
              </h1>
            </div>
            <input
              className="input-class"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />
            
            <input
              className="input-class"
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            
            <input
              id="submit"
              type="submit"
              value="Login"
            />
          </form>
          <div
            className="centre-div"
          >
            <p>
              Don't have an account ?
            </p>
            <Link
              className="links"
              replace={false}
              to="/register"
            >
               Sign up 
            </Link>
          </div>
        </div>,
        "className": "signUp",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <form
              className="auth-form"
              onSubmit={[Function]}
            >
              <div
                className="centre-div"
              >
                <h1>
                   Login 
                </h1>
              </div>
              <input
                className="input-class"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
              />
              
              <input
                className="input-class"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              
              <input
                id="submit"
                type="submit"
                value="Login"
              />
            </form>,
            <div
              className="centre-div"
            >
              <p>
                Don't have an account ?
              </p>
              <Link
                className="links"
                replace={false}
                to="/register"
              >
                 Sign up 
              </Link>
            </div>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="centre-div"
                >
                  <h1>
                     Login 
                  </h1>
                </div>,
                <input
                  className="input-class"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="text"
                  value=""
                />,
                "",
                <input
                  className="input-class"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />,
                "",
                <input
                  id="submit"
                  type="submit"
                  value="Login"
                />,
              ],
              "className": "auth-form",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1>
                     Login 
                  </h1>,
                  "className": "centre-div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Login ",
                  },
                  "ref": null,
                  "rendered": " Login ",
                  "type": "h1",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-class",
                  "id": "email",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Email",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-class",
                  "id": "password",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "submit",
                  "type": "submit",
                  "value": "Login",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Don't have an account ?
                </p>,
                <Link
                  className="links"
                  replace={false}
                  to="/register"
                >
                   Sign up 
                </Link>,
              ],
              "className": "centre-div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Don't have an account ?",
                },
                "ref": null,
                "rendered": "Don't have an account ?",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": " Sign up ",
                  "className": "links",
                  "replace": false,
                  "to": "/register",
                },
                "ref": null,
                "rendered": " Sign up ",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
